chore: Drop Node 20 support, add Node 22/24 prerequisites (#1916)
SAP/ai-sdk-js: README.md

- [Code of Conduct](#code-of-conduct)
- [Licensing](#licensing)

## Prerequisites

Node.js 22 or 24 (LTS) is required.

> Node 20 reached end-of-life on 2026-04-30 and is no longer supported.

Reviewer: [pp] 
```suggestion
Only Node.js LTS and Active LTS are supported.
```

> ### Using the SDK in a CommonJS project (experimental)

Reviewer: ```suggestion
### Using the SDK in a CommonJS project
```

This SDK is published as ESM.

> On Node 22+, CommonJS projects can load it directly with synchronous `require()` — no dynamic `import()` needed:

Reviewer: [pp]
```suggestion
This SDK is published as ESM-only.
CommonJS projects can load it with the synchronous `require()` function (experimental) or asynchronous `import()` function:
```

- [Code of Conduct](#code-of-conduct)
- [Licensing](#licensing)

## Prerequisites

Node.js 22 or 24 (LTS) is required.
Node 20 reached end-of-life on 2026-04-30 and is no longer supported.

### Using the SDK in a CommonJS project (experimental)

This SDK is published as ESM.
On Node 22+, CommonJS projects can load it directly with synchronous `require()` — no dynamic `import()` needed:

```js
const { AzureOpenAiChatClient } = require('@sap-ai-sdk/foundation-models');
```

> **Note:** This relies on Node 22's built-in `require(ESM)` support and is considered experimental.

> > On Node 20, use `await import('@sap-ai-sdk/foundation-models')` instead.

Reviewer: Node 20 is not supported anymore. Nevertheless, it wouldn't hurt to mention it as a way to avoid experimental node features.
```suggestion
```
